x/mongo/driver/topology/connection.go

```diff
@@ -712,13 +713,30 @@ func (c *Connection) ID() string {
 	return c.id
 }
 
+// ServerConnectionID returns the server connection ID of this connection.
+func (c *Connection) ServerConnectionID() *int32 {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
```

Comment:
Calling `ServerConnectionID` on a `Connection` currently doesn't do any locking. Is there a possible race condition that requires us to synchronize calls to `ServerConnectionID`?